src/test/java/org/prebid/server/bidder/adrino/AdrinoBidderTest.java

```diff
@@ -0,0 +1,299 @@
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+    }
+
+    @Test
+    public void makeHttpRequestShouldReturnOneHttpRequestForTowImpsWhenImpsHasSameSourceId() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(Imp.builder()
+                                .xNative(Native.builder().build())
+                                .ext(mapper.valueToTree(
+                                        ExtPrebid.of(null, ExtImpAdrino.of("test"))))
+                                .build(),
+                        Imp.builder()
+                                .xNative(Native.builder().build())
+                                .ext(mapper.valueToTree(
+                                        ExtPrebid.of(null, ExtImpAdrino.of("test"))))
+                                .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = adrinoBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+    }
+
+    @Test
+    public void makeBidsShouldReturnBidWithoutErrors() throws JsonProcessingException {
+        // given
+        final String response = mapper.writeValueAsString(BidResponse.builder()
+                .cur("PLN")
+                .seatbid(singletonList(SeatBid.builder()
+                        .bid(singletonList(Bid.builder().id("impId").impid("test-imp-id").price(BigDecimal.TEN)
+                                .adm("{json response string}")
+                                .crid("test-creative-id")
+                                .ext(mapper.valueToTree(
+                                        ExtPrebid.of(ExtBidPrebid.builder().type(BidType.xNative).build(), null)))
+                                .mtype(4)
+                                .build()))
+                        .build()))
+                .build());
+        BidRequest bidRequest = BidRequest.builder()
+                .id("test-request-id")
+                .imp(singletonList(Imp.builder()
+                        .id("test-imp-id")
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpAdrino.of("123456"))))
+                        .xNative(Native.builder()
+                                .request("{json string}")
+                                .ver("1.2")
+                                .build())
+                        .build()))
+                .site(Site.builder()
+                        .publisher(Publisher.builder()
+                                .id("1")
+                                .build())
+                        .page("some-page-url")
+                        .build())
+                .device(Device.builder()
+                        .w(1920)
+                        .h(800)
+                        .build())
+                .build();
```

Comment:
Since you are not using `BidRequest` to make bids, this object is not needed, just pass `null`. Same in other places.